### PR TITLE
Add Mastery tooltip for an unallocated node when comparing trees

### PR DESCRIPTION
**Pull request by pHiney**
Fixes # Allocated Mastery not showing in the tooltip for the right hand tree when comparing 

### Description of the problem being solved:
From this :
image

To this:
image

### Steps taken to verify a working solution:
- Without compare turned on, hover over everything and ensure the correct information is shown (compare against live PoB) and normal colour.
- With compare turned on, hover over everything and ensure the correct information is shown (compare against live PoB) and normal colour.
- With compare turned on, check that the green mastery node's tooltip reflects the allocated mastery and green in colour.
- With compare turned on, Check Masteries that are allocated in both trees are correct and normal colour.
- With compare turned on and comparing against a tree without the target Mastery being allocated (eg: A3 vs A1 in example), confirm tooltip is correct and normal colour.

### Link to a build that showcases this PR:
https://pastebin.com/bcYQQYg0

Build should default to 'A3' tree. The 'A6' tree is first in the list, so will auto select when enabling compare.
image
image

pH

### CODE REVIEW

**Comment by pHiney** [comment]
### Added showing differences for when the selected Mastery nodes are different
Example 
image

The reverse of the above example. The right side tree is always green.
image

File: src/Classes/PassiveTreeView.lua [hunk 1]
```diff
@@ -932,28 +932,58 @@ function PassiveTreeViewClass:AddNodeTooltip(tooltip, node, build)
...
 	end
 
-	if node.sd[1] and node.allMasteryOptions then
+	-- If node is a Mastery node, check if compare tree is on
+	-- If so, check if the left hand tree is unallocated, but the right hand tree is allocated.
+	-- If so, set the node variable to be the node element from the right hand tree and change the Mastery color
+	-- Then continue processing as normal
+	local masteryColor = ""
+	local mNode = node
+	local rightNodeAlloc = self.compareSpec and self.compareSpec.nodes[node.id].alloc or false
```
Comment (LocalIdentity) [inline src/Classes/PassiveTreeView.lua:941]: Change `rightNodeAlloc` to a more apt name like `compareNode` used above in the code or `compareTree`

File: src/Classes/PassiveTreeView.lua [hunk 2]
```diff
@@ -932,28 +932,58 @@ function PassiveTreeViewClass:AddNodeTooltip(tooltip, node, build)
...
+	if node.type == "Mastery" then
+		if not node.alloc and rightNodeAlloc then
+			mNode = self.compareSpec.nodes[node.id]
+			masteryColor = colorCodes.DEXTERITY
+		end
+		-- If allocated on Left, but not Right, match the colour of the other nodes.
+		if node.alloc and not rightNodeAlloc then
+			masteryColor = colorCodes.STRENGTH
```
Comment (LocalIdentity) [inline src/Classes/PassiveTreeView.lua:949]: This sets all masteries in a build without more than 1 tree to have all contained text as red

**Review by LocalIdentity — CHANGES_REQUESTED** [review]
A few small fixes are needed

File: src/Classes/PassiveTreeView.lua (hunk 1 above)
Comment (pHiney) [reply src/Classes/PassiveTreeView.lua:941]: Funny, that's what it was before i changed it. I was trying to stick left and right trees.

File: src/Classes/PassiveTreeView.lua (hunk 2 above)
Comment (pHiney) [reply src/Classes/PassiveTreeView.lua:949]: Not sure what you trying to show me here ?

Comment (pHiney) [reply src/Classes/PassiveTreeView.lua:949]: solved

**Review by LocalIdentity — APPROVED** [review]

**Pull request merged**